hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+The EDN block will move the first 32 bits from the returned `genbits` bus, and hold on to the remaining data until another request ask for more data. 
+Further, if data is not consumed immediately, the interface to the CSRNG will indicate back pressure to the CSRNG block.
+Each of the four interfaces can request data such that any `genbits` bus return can supply any requesting peripheral block.
+
+A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
+This means that the EDN block will request for that port and is paced by a configurable timer.
+Data will be delivered continuously on the bus for any snooping participant to consume.
+
+Commands to the CSRNG block can be generated by either firmware or hardware.
+For firmware generated commands, a TLUL set of registers will allow any CSRNG application interface command to be sent to CSRNG.
+For hardware generated commands, there are two modes that allow this.
+One is the auto request mode, where two FIFOs are provided to send commands and additional data. 
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOS are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+There is a configurable register that sets the number of requests between reseeds.
+This register determines how many `generate` commands are sent before a `reseed` command is sent.
+The process of sending these two commands will repeat forever until the module is reset or the enable bit is cleared.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+
+## Compatibility
+Commands generated toward the CSRNG application interface must conform to the CSRNG application interface specification.
+
+
+# Theory of Operations
+
+The first step that must be done to enable the EDN block is to configure the  {{< regref "CTRL" >}} register.
+This will enable the block in general, and then determine which of three modes are to be enabled.
+Broadcast mode can be selected independent of auto request mode or boot request mode.
+A choice between auto request mode and boot request mode should be made.
+
+Once a CSRNG command request mode has been configured, the EDN block will start requesting to the CSRNG block.
+Data will be returned from the specific CSRNG instance that the EDN block is connected to.
+It is assumed that the EDN block has sent an `instantiate` application command before sending any `generate` application commands.
+
+Any peripheral module that is on a request/ack bus should expect the EDN block to fetch `genbits` only when requested.
+Conversely, any peripheral module that is on a broadcast bus should expect the EDN block to fetch `genbits` continuously without a request.
```

Comment:
Again, I'm not sure this text is necessary, and it's not even quite right: with the topology in the diagram above, `edn1` fetches data from CSRNG without requests from `ep0`.

I think you're trying to say something like "The EDN fetches data from the CSRNG when it needs to do so, and not otherwise.", but I'm not sure whether this needs writing down! Is the point really that if the EDN has a broadcast bus, it will fetch data from the CSRNG, even if other clients are not requesting it?

Reply:
(I'm seeing a minimalist trend in these comments)
The key point is that the broadcast consumers receive updates without an explicit request.  I can clarify this.

Reply:
> (I'm seeing a minimalist trend in these comments)

Well, it's a spec, so it should be as minimalist as possible, in my book!

Reply:
Perhaps there should be a "Strunk and White" for technical specifications.  (Was "The Elements of Style" ever popular in the U.K., or just the U.S.?  Perhaps to say it has been "popular" in the U.S. is also an exaggeration, though I did once see that Bjarne Stroustrup quoted it in a now-very-dated C++ reference guide.  So it has some international appeal).

Reply:
Whole section has been rewritten.